src/listeners/DialogueBoxController.hx

```diff
@@ -186,11 +198,38 @@ class DialogueBoxController {
 		if (isTalking && Key.isPressed(Key.SPACE)) {
 			if (textState == TypingText) {
 				dialogueText.text = applyTextAttributes(currentText);
-				textState = DialogueBoxState.WaitingForContinue;
+   				textState = DialogueBoxState.WaitingForContinue;
 			} else if (textState == WaitingForContinue) {
 				eventBus.publishEvent(new NextLine());
 			}
 		}
+		if (textState == WaitingForSkillSelection){
```

Comment:
So I'd question putting this here and adding the extra state because the select screen isn't actually a dialgoue box it's just looks like one. I like the event idea. Maybe just add an update function to the randomSelect and move this key logic there. And call that update from the playScene if the randomSelect object isn't null.